packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -139,37 +141,42 @@ export const prepareStatusManager = zone => {
        *
        * @param {NobleAddress} address
        * @param {bigint} amount
+       * @throws {Error} if a pending settlement was not found for the address + amount
        */
       settle(address, amount) {
-        const key = makePendingTxKey(address, amount);
-        const pending = pendingTxs.get(key);
-
+        const pending = statusManager.lookupPending(address, amount);
         if (!pending.length) {
-          throw makeError(`No unsettled entry for ${q(key)}`);
+          throw makeError(`No unsettled entry for ${q([address, amount])}`);
         }
 
         const pendingCopy = [...pending];
         pendingCopy.shift();
         // TODO, vstorage update for `TxStatus.Settled`
-        pendingTxs.set(key, harden(pendingCopy));
+
+        const key = makePendingTxKey(address, amount);
+        if (pendingCopy.length) {
+          return pendingTxs.set(key, harden(pendingCopy));
+        }
+        return pendingTxs.delete(key);
       },
 
       /**
-       * Lookup all pending entries for a given address and amount
+       * Lookup all pending entries for a given address and amount.
        *
        * @param {NobleAddress} address
        * @param {bigint} amount
-       * @returns {PendingTx[]}
+       * @returns {PendingTx[]} pendingTxs or empty array if nothing is found
        */
       lookupPending(address, amount) {
         const key = makePendingTxKey(address, amount);
         if (!pendingTxs.has(key)) {
-          throw makeError(`Key ${q(key)} not yet observed`);
+          return harden([]);
         }
         return pendingTxs.get(key);
       },
     },
   );
+  return statusManager;
```

Comment:
nit: i found the direct `return` a little clearer. This way I was looking for how `statusManager` is used